File: models/assetmanagement/AssetSubCategory.js
import mongoose from "mongoose";

const AssetSubCategorySchema = new mongoose.Schema(
{
    hospital_id: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branch_id: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    category_id: { type: mongoose.Schema.Types.ObjectId, ref: "AssetCategory", required: true },

    name: { type: String, required: true },
    description: String

},
{ timestamps: true }
);

export default mongoose.model("AssetSubCategory", AssetSubCategorySchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AssetSubCategory.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "asset_subcategories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AssetSubCategory extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private AssetCategory category;

    @Column(nullable = false)
    private String name;

    private String description;
}
